src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql.java

```diff
@@ -0,0 +1,130 @@
+
+/**
+ * Data migration class for course entity.
+ */
+@SuppressWarnings("PMD")
+public class DataMigrationForCourseEntitySql extends
+        DataMigrationEntitiesBaseScriptSql<teammates.storage.entity.Course, teammates.storage.sqlentity.BaseEntity> {
+
+    public static void main(String[] args) {
+        new DataMigrationForCourseEntitySql().doOperationRemotely();
+    }
+
+    @Override
+    protected Query<Course> getFilterQuery() {
+        return ofy().load().type(teammates.storage.entity.Course.class);
+    }
+
+    @Override
+    protected boolean isPreview() {
+        return false;
+    }
+
+    /*
+     * Sets the migration criteria used in isMigrationNeeded.
+     */
+    @Override
+    protected void setMigrationCriteria() {
+        // No migration criteria currently needed.
+    }
+
+    @Override
+    protected boolean isMigrationNeeded(Course entity) {
+        return true;
+    }
+
+    @Override
+    protected void migrateEntity(Course oldCourse) throws Exception {
+        teammates.storage.sqlentity.Course newCourse = createCourse(oldCourse);
+        // TODO: add shutdown hook to save the entity
+        // Runnable shutdownScript = () -> { cascadeDelete(newCourse)};
+        // Runtime.getRuntime().addShutdownHook(new Thread(shutdownScript));
+
+        migrateCourseEntity(newCourse);
+        // verifyCourseEntity(newCourse);
+        // markOldCourseAsMigrated(courseId)
+        // Runtime.getRuntime().removeShutDownHook(new Thread(shutdownScript));
+    }
+
+    private teammates.storage.sqlentity.Course createCourse(Course oldCourse) {
+        teammates.storage.sqlentity.Course newCourse = new teammates.storage.sqlentity.Course(
+                oldCourse.getUniqueId(),
+                oldCourse.getName(),
+                oldCourse.getTimeZone(),
+                oldCourse.getInstitute());
+        newCourse.setDeletedAt(oldCourse.getDeletedAt());
+        // newCourse.setCreatedAt(oldCourse.getCreatedAt());
+        saveEntityDeferred(newCourse);
+        return newCourse;
+    }
+
+    private teammates.storage.sqlentity.Section createSection(teammates.storage.sqlentity.Course newCourse,
+            String sectionName) {
+        return new teammates.storage.sqlentity.Section(newCourse, sectionName);
+    }
+
+    private teammates.storage.sqlentity.Team createTeam(teammates.storage.sqlentity.Section section, String teamName) {
+        return new teammates.storage.sqlentity.Team(section, teamName);
+    }
+
+    private void migrateCourseEntity(teammates.storage.sqlentity.Course newCourse) {
+        Map<String, teammates.storage.sqlentity.Section> sectionNameToSectionMap = migrateSectionChain(newCourse);
+        System.out.println(sectionNameToSectionMap); // To stop lint from complaining
+        // TODO: Add mirgrateFeedbackChain
+        // migrateFeedbackChain(sectionNameToSectionMap);
+    }
+
+    private Map<String, teammates.storage.sqlentity.Section> migrateSectionChain(
```

Comment:
Could we arrange the methods in order so it'll be clearer, whichever the order. E.g the main "chain" methods, then the helper methods, or vice versa.

Reply:
Sure will do!